pkg/network/encoding/http.go

```diff
@@ -109,11 +112,15 @@ func (e *httpEncoder) GetHTTPAggregationsAndTags(c network.ConnectionStats) (*mo
 }
 
 func (e *httpEncoder) buildAggregations(payload *network.Connections) {
-	aggrSize := make(map[types.ConnectionKey]int)
-	for key := range payload.HTTP {
+	aggrSize := make(map[types.ConnectionKey]int, len(payload.HTTP))
+	dataPoolSize := 0
+	for key, stat := range payload.HTTP {
 		aggrSize[key.ConnectionKey]++
+		dataPoolSize += len(stat.Data)
 	}
 
+	e.dataPool = make([]model.HTTPStats_Data, dataPoolSize)
```

Comment:
do you have to reset poolIndex somewhere too?

Reply:
changed the logic